zrml/swaps/rpc/src/lib.rs

```diff
@@ -205,4 +217,37 @@ where
             })
             .collect()
     }
+
+    async fn get_all_spot_prices(
+        &self,
+        pool_id: PoolId,
+        with_fees: bool,
+        blocks: Vec<NumberFor<Block>>,
+    ) -> RpcResult<BTreeMap<SerdeWrapper<NumberFor<Block>>, Vec<(Asset<MarketId>, Balance)>>> {
+        let api = self.client.runtime_api();
+        let mut res: BTreeMap<SerdeWrapper<NumberFor<Block>>, Vec<(Asset<MarketId>, Balance)>> =
+            BTreeMap::new();
```

Comment:
You don't need this type annotation. Since you basically just copying the annotation from the function's signature, it's best to remove this:

```suggestion
        let mut res = BTreeMap::new();
```